gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,322 @@
+
+  private static final String SCHEMA_DDL =
+      "CREATE TABLE IF NOT EXISTS %s (\n"
+          + "  "
+          + COL_TYPE
+          + " string COMMENT 'Type of information this row represents, marker row or content row',\n"
+          + "  "
+          + COL_GC_RUN_START
+          + " timestamp COMMENT 'GC run start timestamp',\n"
+          + "  "
+          + COL_GC_RUN_ID
+          + " string COMMENT 'GC run-ID',\n"
+          + "  "
+          + COL_CONTENT_ID
+          + " string COMMENT 'Nessie Content.id',\n"
+          + "  "
+          + COL_EXPIRED_CONTENTS
+          + " array<String> COMMENT 'List of expired contents (as a json serialized string), can be empty',\n"
+          + "  "
+          + COL_EXPIRED_AT_REFERENCE_NAME
+          + " string COMMENT 'Name of the reference via which the contentID is visible,'\n"
+          + ")\n"
+          + "USING iceberg\n"
+          + "PARTITIONED BY ("
+          + COL_GC_RUN_START
+          + ")";
+
+  private final SparkSession sparkSession;
+  private final String catalogAndTable;
+  private final String gcRefName;
+  private final StructType schema;
+
+  public IdentifiedResultsRepo(
+      SparkSession sparkSession, String catalog, String gcRefName, String gcTable) {
+    this.sparkSession = sparkSession;
+    this.catalogAndTable = String.format("%s.%s", catalog, gcTable);
+    this.gcRefName = gcRefName;
+    this.schema =
+        checkAndCreateTable(
+            sparkSession, catalog, TableIdentifier.parse(gcTable), catalogAndTable, gcRefName);
+  }
+
+  public StructType getSchema() {
+    return schema;
+  }
+
+  /**
+   * Collect the expired contents for the given run id as {@link IdentifiedResult} .
+   *
+   * @param runId run id of completed identify task.
+   * @return {@link IdentifiedResult} object having the expired contents per content id per
+   *     reference.
+   */
+  public IdentifiedResult collectExpiredContents(String runId) {
+    List<Row> rows = collectExpiredContentsAsDataSet(runId).collectAsList();
+    IdentifiedResult identifiedResult = new IdentifiedResult();
+    ObjectMapper objectMapper = new ObjectMapper();
+    rows.forEach(row -> addRowToIdentifiedResults(identifiedResult, objectMapper, row));
+    return identifiedResult;
+  }
+
+  /**
+   * Collect the expired contents for the given run id as spark dataset.
+   *
+   * @param runId run id of completed identify task.
+   * @return spark dataset of row where each row is having the expired contents per content id per
+   *     reference.
+   */
+  public Dataset<Row> collectExpiredContentsAsDataSet(String runId) {
+    // collect marker row for the given run-id
+    String checkMarkerRow =
+        String.format(
+            "SELECT %s FROM %s WHERE %s = '%s' AND %s = '%s'",
+            COL_GC_RUN_ID,
+            withRefName(catalogAndTable, gcRefName),
+            COL_GC_RUN_ID,
+            runId,
+            COL_TYPE,
+            TYPE_GC_MARKER);
```

Comment:
maybe extract to helper method? It is used in multiple places

Reply:
sorry, where is it repeated? I think arguments are different.

Reply:
I think something like the helper method in icebergs tests which makes it easier to read. Something like

`Dataset<Row> sql(sqlStatement, Object... args)`